### ISSUE TITLE
Using declaration with ‘_’ should be a discard

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#43292]
[Sharplab](https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQMwAJboMLpOgSQBEBLAZwAcB7MgQwCMAbAU3QG9l0vMMoAWdAFkAFAEp0nbhyTdZmAAzoAbrQBO6AProAvOgB2zAO44xAbklzNAOgAqVAMoAXVST0BzMxfQBfL2kwCpJQ0zGLsvkjeQA==)

‘using var _ = ...;’ currently introduces a local in scope and it can be accessed. ‘_.ToString();’ should fail.

Tagging @chsienki

**Comment by chsienki** [comment]
I thought we had another issue tracking this already, but I can't seem to find it. We've talked a few times about allowing using discards, but I don't think we ever came to a concrete conclusion.

It seems like a fairly innocuous thing to allow, but we should check with LDM first.

**Comment by jcouv** [comment]
Looking at the LDM notes, I see that we did get a decision: https://github.com/dotnet/csharplang/blob/master/meetings/2018/LDM-2018-12-03.md#discussion

Please scan those notes to check if anything else was left un-implemented.

**Comment by Joe4evr** [comment]
This was also reported on csharplang over a year ago: dotnet/csharplang#2235

**Comment by jcouv** [comment]
@chsienki I don't know what you think, but suspect that fixing this will be rather involved. Moving out to 16.9.

**Comment by Sergio0694** [comment]
Fixing this would also be a nice workaround for https://github.com/dotnet/csharplang/issues/2235.
That is, it would allow us to at least do this:

```csharp
using var _ = Expr1();
using var _ = Expr2();
using var _ = Expr3();

// Code...
```

And not have a whole bunch of indentation due to having to fallback to `using (Expr()) { }` 🥲

**Comment by ririshi** [comment]
This has been on the backlog for a pretty good while now. The behaviour already exists for `out var _`, would it be very involved to translate this to using declarations?

I use this pattern quite often for things like locking semaphores and scoped logging. Being forced to create unused locals is making this pattern look unnecessarily messy.

**Comment by CyrusNajmabadi** [comment]
@Ririshi see https://gist.github.com/CyrusNajmabadi/0d4559b4528792e55edec58974cefc93
